File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int zero = 0;
        for(int i=0; i<6; i++){         //지워진 숫자의 갯수를 카운트
            if(lottos[i] == 0){
                zero++;
            }
        }
        boolean flag[] = new boolean[6];
        int cnt = 0;
        for(int i=0; i<6; i++){
            for(int j=0; j<6; j++){
                if(!flag[i]){
                    if(lottos[i] == win_nums[j]){   //남은 숫자 중 맞은 갯수를 카운트
                    cnt++;
                        flag[i] = true;
                    }
                }
            }
        }
        System.out.println(zero + "/" + cnt);   //지워진 숫자가 다 맞을경우 = max, 다 틀릴경우 = min
        int max = 7 - zero - cnt;
        if(max >= 6){
            max = 6;
        }
        int min = 7 - cnt;
        if(min >= 6){
            min = 6;
        }
        int[] answer = {max , min};
        return answer;
    }
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include<iostream>
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    sort(lottos.begin(), lottos.end());
    sort(win_nums.begin(), win_nums.end());
    int zero = 0;
    int correct = 0;
    for(int i=0; i<6; i++){
        if(lottos[i] == 0){
            zero++;
        }
        else{
            for(int j=0; j<6; j++){
                if(lottos[i] == win_nums[j]){
                    correct++;
                }
            }
        }
    }
    vector<int> answer;
    if(7-(zero+correct)<6){
        answer.push_back(7-(zero+correct));
    }
    else{
        answer.push_back(6);
    }
    if(7-correct <6){
        answer.push_back(7-correct);
    }
    else{
        answer.push_back(6);
    }
    return answer;
}